### PR TITLE
Pr/coex stats

### PR DESCRIPTION
**Pull request by ankuns**

### CODE REVIEW

File: src/platform/coex/nrf_802154_wifi_coex.h [hunk 1]
```diff
@@ -53,6 +55,17 @@ extern "C" {
  * to assert pins and respond to pin state changes.
  */
 
+/**@brief Enumeration representing state of request to PTA */
+typedef enum
+{
+    /** @brief No request is being made to PTA. */
+    WFFI_COEX_REQUEST_STATE_NO_REQUEST = 0,
```
Comment (jciupis) [inline src/platform/coex/nrf_802154_wifi_coex.h]: typo: `WFFI`

File: src/rsch/nrf_802154_rsch.h [hunk 2]
```diff
@@ -252,6 +252,15 @@ bool nrf_802154_rsch_timeslot_is_requested(void);
...
 bool nrf_802154_rsch_prec_is_approved(rsch_prec_t prec, rsch_prio_t prio);
 
+/**
+ * @brief Gets value of approved RSCH precondition.
+ *
+ * @param[in] prec    RSCH precondition to be got.
+ *
+ * @return Approved priority level of the given precondition.
+ */
+rsch_prio_t nrf_802154_rsch_approved_prec_get(rsch_prec_t prec);
```
Comment (jciupis) [inline src/rsch/nrf_802154_rsch.h]: I think this function is unnecessary and `nrf_802154_rsch_prec_is_approved` could be used instead

File: src/nrf_802154_core.c [hunk 3]
```diff
@@ -1506,6 +1506,12 @@ void nrf_802154_trx_receive_frame_received(void)
             }
             else
             {
+                if (nrf_802154_rsch_approved_prec_get(RSCH_PREC_COEX) <
```
Comment (jciupis) [inline src/nrf_802154_core.c]: Existing API could be used instead:
```
if (!nrf_802154_rsch_prec_is_approved(RSCH_PREC_COEX, min_required_rsch_prio(RADIO_STATE_TX_ACK)))
```

File: src/rsch/nrf_802154_rsch.c [hunk 4]
```diff
@@ -692,23 +699,56 @@ void nrf_802154_clock_hfclk_ready(void)
...
+
     nrf_802154_log_function_exit(NRF_802154_LOG_VERBOSITY_LOW);
 }
+
+void nrf_802154_wifi_coex_request_changed(
+    nrf_802154_wifi_coex_request_state_t curr_request_state,
+    nrf_802154_wifi_coex_request_state_t prev_request_state,
+    bool                                 grant_state)
+{
+    (void)prev_request_state;
```
Comment (jciupis) [inline src/rsch/nrf_802154_rsch.c]: Is this line needed? `prev_request_state` is used in the if clause below

Comment (ankuns) [reply src/rsch/nrf_802154_rsch.c]: leftover, I'll fix

**Pull request merged**